modules/autotagging-commons/common/src/main/java/org/opensearch/rule/RuleUtils.java

```diff
@@ -112,4 +114,32 @@ public static Rule composeUpdatedRule(Rule originalRule, UpdateRuleRequest reque
             Instant.now().toString()
         );
     }
+
+    /**
+     * Builds a flattened map of attribute filters from a {@link Rule}.
+     * This method reformats nested or prioritized subfields (e.g., values containing "|" for sub-attributes)
+     * into top-level attribute keys. For example, an attribute "principal" with value "username|admin" will
+     * become "principal.username" -> "admin" in the resulting map. Attributes without prioritized subfields
+     * remain unchanged.
+     * The resulting map is structured to make querying rules from the index easier.
+     * @param rule the rule whose attributes are to be flattened
+     */
+    public static Map<String, Set<String>> buildAttributeFilters(Rule rule) {
```

Comment:
I think since the return value here will be used inside `GetRuleRequest`, does it make sense to move this method into that class.